addons/mail/models/mail_partner.py

```diff
@@ -0,0 +1,132 @@
+from datetime import datetime
+from odoo import api, fields, models, tools, SUPERUSER_ID
+
+class MailPartnerMixin(models.AbstractModel):
```

Comment:
I put a general remark here about the naming. I think most, if not all, of the methods of this mixin should be private (begin with an '_'). It makes no sense to make them available for external API. The naming could also be a bit improved. Since a few weeks some renaming has been performed in mail module. message prefix methods related to the post and routing mechanism; notify for the notfiication mechanism for exemple. I don't know if we could find some prefix different from message here to avoid confusion with other methods form mail.thread.